src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Sharing/SpatialCoordinateSystemManager.cs

```diff
@@ -220,6 +252,13 @@ private async void OnLocalizeMessageReceived(SocketEndpoint socketEndpoint, stri
             }
 
             await RunLocalizationSessionAsync(localizer, settings, participant);
+
+            // Ensure that the participant's fully-localized state is sent before sending the LocalizationComplete command (versus waiting
+            // for the next Update). This way the remote peer receives the located state of the participant before they receive the notification
+            // that this localization session completed.
+            participant.CheckForStateChanges();

```

Comment:
>CheckForStateChanges [](start = 24, length = 20)

nit: this function name is misleading. SendStateInformation may be a more appropriate naming, technically this doesn't check for a state change it just reports the current state.

Reply:
Agreed, renamed this to EnsureStateChangesAreBroadcast

---
In reply to: [295546110](https://github.com/microsoft/MixedReality-SpectatorView/pull/26#discussion_r295546110) [](ancestors = 295546110)